persistence/persistence-api/src/main/java/org/cbioportal/persistence/MolecularDataRepository.java

```diff
@@ -6,14 +6,15 @@
 import org.cbioportal.model.GenericAssayMolecularAlteration;
 import org.cbioportal.model.GenesetMolecularAlteration;
 import org.springframework.cache.annotation.Cacheable;
+import org.cbioportal.model.MolecularProfileSamples;
 
 public interface MolecularDataRepository {
 
     @Cacheable(cacheNames = "GeneralRepositoryCache", condition = "@cacheEnabledConfig.getEnabled()")
-    String getCommaSeparatedSampleIdsOfMolecularProfile(String molecularProfileId);
+    MolecularProfileSamples getCommaSeparatedSampleIdsOfMolecularProfile(String molecularProfileId);
 
     @Cacheable(cacheNames = "GeneralRepositoryCache", condition = "@cacheEnabledConfig.getEnabled()")
-    List<String> getCommaSeparatedSampleIdsOfMolecularProfiles(List<String> molecularProfileIds);
+    List<MolecularProfileSamples> getCommaSeparatedSampleIdsOfMolecularProfiles(List<String> molecularProfileIds);
```

Comment:
I've noticed many(maybe all?) places in the code where you convert `List<MolecularProfileSamples>` to `Map<String, MolecularProfileSamples>`.  Maybe the signature should be updated to support the return of a Map?